ckanext/dia_theme/templates/header_base.html

```diff
@@ -120,7 +120,7 @@ <h1>
                 <div class="site-search__visibility hidden" id="site-search" data-rua-toggle="slide" style="display: none;" aria-expanded="false">
                     <div class="container">
                         <div class="site-search__inner">
-                            <form class="search-form" id="dataset-search" action="{% url_for controller='package', action='search' %}" method="get" data-module="select-switch">
+                            <form class="search-form" id="dataset-search" action="{% url_for controller='dataset' if h.ckan_version().split('.')[1] | int >= 9 else 'package', action='search' %}" method="get" data-module="select-switch">
```

Comment:
I know this is done in other modules and recommended in the upgrade docs... what happens when this code lives through to ckan 3.0? Is there an implicit idea that anything that needs this will also need to be updated in 3.0? I have no idea what ckan's versioning strategy is.

Reply:
I did a little playing with this, looks like standard Python string comparison should work here:
```
>>> version = '2.9.3'
>>> version >= '2.9'
True
>>> version = '3.0.0'
>>> version >= '2.9'
True
>>> version = '2.7.7'
>>> version >= '2.9'
False
```
Can you see anything wrong with using this comparison instead of the `version.split('.')[1] | int >= 9`?

Reply:
I though it was a bit ugly myself, was just following instructions in my first week. 😉 I suppose the question is does python2 operate in the same way? I'll do some playing around and see if I can make it cleaner.

Reminds me of the whole skipping Windows 9 because of all the `9.*` matches that existed for 95/98.

Reply:
After taking a look it just uses lexicographic ordering. So it falls over when you have a two digit section.
```
'2.10.0' > '2.9'
```

I could probably create a helper within the plugin or tighten it up so it includes comparing the major version. 
```
tuple(int(x) for x in h.ckan_version().split('.')) < (2, 9, 0)
```

Reply:
Looks like they do have a helper. https://docs.ckan.org/en/2.9/extensions/plugins-toolkit.html#ckan.plugins.toolkit.ckan.plugins.toolkit.check_ckan_version

Reply:
Great, same helper exists as far back as ckan 2.6 if not further :+1: